webapp/channels/src/components/save_button.tsx

```diff
@@ -17,65 +17,55 @@ type Props = {
     extraClasses?: string;
 }
 
-export default class SaveButton extends React.PureComponent<Props> {
-    public static defaultProps: Partial<Props> = {
-        btnClass: '',
-        defaultMessage: (
-            <FormattedMessage
-                id='save_button.save'
-                defaultMessage='Save'
-            />
-        ),
-        disabled: false,
-        extraClasses: '',
-        savingMessage: (
-            <FormattedMessage
-                id='save_button.saving'
-                defaultMessage='Saving'
-            />
-        ),
-    };
-
-    public render() {
-        const {
-            saving,
-            disabled,
-            savingMessage,
-            defaultMessage,
-            btnClass,
-            extraClasses,
-            ...props
-        } = this.props;
-
-        let className = 'btn';
-        if (!btnClass) {
-            className += ' btn-primary';
-        }
+const SaveButton = ({
+    disabled = false,
+    btnClass = '',
+    extraClasses = '',
+    savingMessage = (
+        <FormattedMessage
+            id='save_button.saving'
+            defaultMessage='Saving'
+        />
+    ),
+    defaultMessage = (
+        <FormattedMessage
+            id='save_button.save'
+            defaultMessage='Save'
+        />
+    ),
+    saving,
+    ...props
+}: Props) => {
+    let className = 'btn';
+    if (!btnClass) {
+        className += ' btn-primary';
+    }
 
-        if (!disabled || saving) {
-            className += ' ' + btnClass;
-        }
+    if (!disabled || saving) {
+        className += ' ' + btnClass;
+    }
 
-        if (extraClasses) {
-            className += ' ' + extraClasses;
-        }
+    if (extraClasses) {
+        className += ' ' + extraClasses;
+    }
 
-        return (
-            <button
-                type='submit'
-                data-testid='saveSetting'
-                id='saveSetting'
-                className={className}
-                disabled={disabled}
-                {...props}
+    return (
+        <button
+            type='submit'
+            data-testid='saveSetting'
+            id='saveSetting'
+            className={className}
+            disabled={disabled}
+            {...props}
+        >
+            <LoadingWrapper
+                loading={saving}
+                text={savingMessage}
             >
-                <LoadingWrapper
-                    loading={saving}
-                    text={savingMessage}
-                >
-                    <span>{defaultMessage}</span>
-                </LoadingWrapper>
-            </button>
-        );
-    }
-}
+                <span>{defaultMessage}</span>
```

Comment:
is there a reason we use span here, as <FormattedMessage creates span on its own

Reply:
I have not changed anything. This `span` was already there.  `defaultMessage` comes from props so if  the user assigns some other thing to defaultMessage instead of `FormattedMessage` then there will be no `span` element.

Reply:
Yes can you verify if there is any difference between with and without span here. I would like to remove if it doesnt make any difference UI wise